### ISSUE TITLE
Trilateration function returns an incorrect result

### ISSUE THREAD

**Issue by casc2** [mrJean1/PyGeodesy#33]
I ran into some odd results while attempting to locate a point using the trilaterate function (calculated points were almost thousands of meters far from their actual locales), so I decided to check these results against another [Python implementation](https://gis.stackexchange.com/questions/66/trilateration-using-3-latitude-longitude-points-and-3-distances) based on the algorithm found in [Movable Type website](https://www.movable-type.co.uk/scripts/latlong-vectors.html), the JavaScript equivalent to this repository. It turned out that the "same" algorithm was returning points with centimeters precision.

Comparing the equations in the link above to the ones found in the [sphericalNvector.py](https://github.com/mrJean1/PyGeodesy/blob/723f27f6a7da98630e722c4c1241b750ae696182/pygeodesy/sphericalNvector.py) file, the following lines seems a little off to me:
https://github.com/mrJean1/PyGeodesy/blob/723f27f6a7da98630e722c4c1241b750ae696182/pygeodesy/sphericalNvector.py#L1086-L1087

According to the website, it should be something like this instead:
```python
x = fsum_(d12, -d22, d**2) / (2 * d)  # n1->intersection x- and ...
y = fsum_(d12, -d32, i**2, j**2) / (2 * j) - (x * i / j) # ... y-component
```
After these changes, both algorithms (the one from the link above and the implementation in this project) returned approximately the same points with centimeters precision, as I said earlier.

I tried to open a pull request with this possible fix, but I have no permission to push to this repository. Nonetheless, I willing to do so if you want me to.

**Comment by mrJean1** [comment]
Thank you for reporting and fixing this obvious mistake on my part.

Another issue is that there are no tests for the `trilaterate` function.  Do you have one (or perhaps more) test cases?

**Comment by casc2** [comment]
Thanks for the quick response! I'm happy to contribute with your library!

I do have a good amount of test cases for trilateration, since the project I'm working on deals with a database (~2.000 entries) of geographic coordinates. Although, I'm not sure I'll be able to collaborate  in due time with your test suite by implementing the test functions. Graduation and work are keeping me quite busy these last weeks!

Do you have any plans to release a new version with this fix in the next few days? I'd like to continue using PyGeodesy in my project, 'cause it's such a useful and well documented Python library (instead of using some separate functions).

**Comment by mrJean1** [comment]
No rush, I plan to release an update with fixes for 2 issues, this and #31.

No need to write test functions, just some `trilaterate` data with expected, verified results would be great.  Several weeks or months from now is fine, for another release scheduled then.

**Comment by casc2** [comment]
Alright then! I'll send you the data as soon as possible.

**Comment by mrJean1** [comment]
Just found and added a `trilaterate` test case to `pygeodesy/test/testVectorial.py`:

```
if hasattr(LatLon, 'trilaterate'):
    # <https://GitHub.com/chrisveness/geodesy/blob/master/test/latlon-nvector-spherical-tests.js>
    p = LatLon(37.418436, -121.963477)
    t = p.trilaterate(265.710701754, LatLon(37.417243, -121.961889), 234.592423446,
                                     LatLon(37.418692, -121.960194), 54.8954278262)
    self.test('trilaterate', t, '37.419078°N, 121.960579°W')
    self.test('trilaterate', isinstance(t, LatLon), True)
```

The test passes with your fixes.  Thank you again.

**Comment by casc2** [comment]
Great! I can provide more test cases in the next weeks, If you're still interested.

Always a pleasure to help!

**Comment by mrJean1** [comment]
Yes, please do.  Tests are critical.

**Comment by mrJean1** [comment]
`PyGeodesy 19.6.14` is out and includes the `trilaterate` (and another) fix.  Please close this issue if you approve.

**Comment by casc2** [comment]
The following test case raises an error:
```python
point_1 = LatLon(-8.068912, -34.888699)

bts_1 = LatLon(-8.068361, -34.892722)
bts_2 = LatLon(-8.075917, -34.894611)
bts_3 = LatLon(-8.076361, -34.908000)

dist_1 = bts_1.distanceTo(point_1)
dist_2 = bts_2.distanceTo(point_1)
dist_3 = bts_3.distanceTo(point_1)

bts_1.trilaterate(dist_1, bts_2, dist_2, bts_3, dist_3)
```
Error message:
`ValueError: no trilaterate for LatLon(08°04′06.1″S, 034°53′33.8″W), LatLon(08°04′33.3″S, 034°53′40.6″W), LatLon(08°04′34.9″S, 034°54′28.8″W) at 447.1270579647369, 1015.0628081200015, 2280.63761581466`

Yet, running the same input in the `trilaterate` implementation based on the JS source code returns the exact coordinates of `point_1`.

It probably has something to do with the changes in the `z` coordinate equation, since this coordinate is partially ignored in the [JS implementation](https://github.com/chrisveness/geodesy/blob/454b5dadf32ecb7e965f1fe410734ac2e32c644b/latlon-nvector-spherical.js#L684-L685):
https://github.com/mrJean1/PyGeodesy/blob/d59042010f872c60413ff55f63138e292211217b/pygeodesy/sphericalNvector.py#L1089-L1091

**Comment by mrJean1** [comment]
Thank you for testing and reporting the issue.  `PyGeodesy 19.6.17` includes the fix and the test case.

Perhaps, there should be an option to include the `z` component, with default ignore `z` as is now.

Also, the `height` is still computed as the mean of the heights of the 3 points.  Interpolating the height using IDW (Inverse Distance Weighting) would be better.

**Comment by casc2** [comment]
I believe the improvements that you're suggesting about the `z` component and `height` make perfect sense. Maybe you should consider them for the next release. For the time being, `trilaterate` is working just fine with all my test cases! Thank you for the quick fix!

**Comment by mrJean1** [comment]
Thank you for your comments and confirmation and for closing the issue.

Also, both the `z` and `height` ideas are already included in `PyGeodesy 19.6.26`.  Function `sphericalNvector.trilaterate` has a new keyword argument `useZ=False` and the `height` is interpolated using IDW, iff not specified.

**Comment by mrJean1** [comment]
Another problem with `sphericalNvector.trilaterate` has been reported in issue #43, **Trilaterate by area**.  That bug has been fixed in [PyGeodesy 20.9.23](https://github.com/mrJean1/PyGeodesy/).

That release also includes a new method `trilaterate5` for all `LatLon` classes (with restrictions for `-Nvector-based`), more details are in the [documentation](https://mrjean1.github.io/PyGeodesy/docs/pygeodesy.ellipsoidalBase.LatLonEllipsoidalBase-class.html#trilaterate5).
